ui/app/routes/vault/cluster/secrets/backend/secret-edit.js

```diff
@@ -75,9 +75,22 @@ export default Route.extend({
 
   beforeModel({ to: { queryParams } }) {
     const secret = this.secretParam();
+    const secretEngine = this.modelFor('vault.cluster.secrets.backend');
     return this.buildModel(secret, queryParams).then(() => {
       const parentKey = parentKeyForKey(secret);
       const mode = this.routeName.split('.').pop();
+      // for kv v2, redirect users from the old url to the new engine url (1.15.0 +)
+      if (secretEngine.type === 'kv' && secretEngine.version === 2) {
+        // if no secret param redirect to the create route
+        // if secret param they are either viewing or editing secret so navigate to the details route
```

Comment:
Wouldn't they just be editing since this is the `secret-edit` route?

Reply:
So I got this from the line [here](https://github.com/hashicorp/vault/blob/main/ui/app/routes/vault/cluster/secrets/backend/secret-edit.js#L185). And when I log mode from cubbyhole "Create" it hits this hook and the mode is "create-root". The routing is hard to following so I literally just went with logging to figure if I needed to catch this path here.

Reply:
Ah okay - agreed, it is hard to follow!